shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -182,6 +183,115 @@
   return canvases;
 }
 
+// Builds a chain of UIViews that applies the mutations described by mutatorsStack.
+//
+// Clips are applied by adding a super view with a CALayer mask. Transforms are applied to the
+// view that's at the head of the chain. For example the following mutators stack [T_1, C_2, T_3,
+// T_4, C_5, T_6] where T denotes a transform and C denotes a clip, will result in the following
+// UIView tree:
+//
+// C_2 -> C_5 -> PLATFORM_VIEW
+// (PLATFORM_VIEW is a subview of C_5 which is a subview of C_2)
+//
+// T_1 is applied to C_2, T_3 and T_4 are applied to C_5, and T_6 is applied to PLATFORM_VIEW.
+//
+// UIView instances used for clip mutations are re-used when possible.
+// For example if the mutators stack above is updated to [T_1, C_7, T_3, T_4, C_8, T_9], we do not
+// remove the previous superviews and re-create new views. Instead, we reuse the superviews that
+// were used to perform clips in the previous example and perform the new clips on the same views.
+//
+// Returns the UIView that's at the top of the generated chain(C_2 for the example above).
+UIView* FlutterPlatformViewsController::ApplyMutators(const MutatorsStack& mutators_stack,
```

Comment:
f2f discussion: we feel like we can make the code easier to follow by doing something like(pseudo code):

```
composite_embedded_views(params) {
  clip_count = count_clips(params);
  if(clip_count != prev_clip_count) {
    detach_chain_from_flutter_view(); // the chain will be attached in SubmitFrame
    restructure_mutator_view_chain(); // make sure we have clip_count UIViews between the platform view and flutter view.
  }
  apply_mutators() // applies the current parameters to the UIView chain
}
```